src/js/actions/menu.js

```diff
@@ -570,6 +570,181 @@ define(function (require, exports) {
         writes: [locks.JS_PREF],
         transfers: [preferencesActions.setPreference]
     };
+    
+    /**
+     * Debug-only method to toggle Descriptor Event logging
+     * 
+     * @param {Object} options
+     * @param {Object} options.toggle
+     * @return {Promise}
+     */
+    var logDescriptor = function (options) {
+        if (!__PG_DEBUG__) {
+            return Promise.resolve();
+        }
+
+        options = _.merge({ toggle: true }, options);
+
+        var preferencesStore = this.flux.store("preferences"),
+            preferences = preferencesStore.getState(),
+            enabled = preferences.get("descriptorLoggingEnabled"),
+            descriptor = require("adapter").ps.descriptor;
+
+        if (options.toggle) {
+            enabled = !enabled;
+        }
+        
+        var blockStyle = "color:gray;",
+            eventStyle = "font-weight:bold";
+        
+        descriptor.__play = descriptor.__play || descriptor.play;
```

Comment:
I think it would be simpler and safer to instead replace the implementation of `_getAsync` and `_batchPlayAsync`. It seems like that might also work a little better with the transaction infrastructure.